File: app/src/main/java/com/locus/books/MainActivity.java
package com.locus.books;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    private EditText searchBox;
    private Button btn;
    public final static String SEARCH_TEXT = "search_text";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        searchBox = (EditText) findViewById(R.id.search_text);
        btn = (Button) findViewById(R.id.search_btn);
        btn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String text = searchBox.getEditableText().toString();
                if (!TextUtils.isEmpty(text)) {
                    Intent intent = new Intent(getApplicationContext(), BooksListActivity.class);
                    intent.putExtra(SEARCH_TEXT, text);
                    startActivity(intent);
                } else {
                    Toast.makeText(getApplicationContext(), "Text Empty", Toast.LENGTH_SHORT).show();
                }
            }
        });
    }
}

File: src/main/scala/com/rxbytes/books/explorer/ui/main/MainActivity.scala
package com.rxbytes.books.explorer.ui.main

import android.os.Bundle
import android.support.v4.app.FragmentActivity
import android.support.v7.app.AppCompatActivity
import macroid.FullDsl._
import macroid._


class MainActivity
  extends AppCompatActivity
    with Contexts[FragmentActivity]
    with Layout
    with IdGeneration {
  self =>

  override def onCreate(savedInstanceState: Bundle) = {
    super.onCreate(savedInstanceState)

    setContentView(layout)
    toolBar foreach setSupportActionBar
    getSupportActionBar.setDisplayHomeAsUpEnabled(true)
    getSupportActionBar.setHomeButtonEnabled(true)

  }
}
